evalml/tests/automl_tests/test_automl.py

```diff
@@ -3860,175 +3115,29 @@ def test_automl_pipeline_params_multiple(AutoMLTestEnv, X_y_regression):
-            assert row["parameters"]["Imputer"]["numeric_impute_strategy"] == "mean"
-            assert row["parameters"]["Imputer_1"]["numeric_impute_strategy"] in {
-                "most_frequent",
-                "mean",
-            }
-            assert row["parameters"]["Random Forest Classifier"]["n_estimators"] in {
-                100,
-                125,
-            }
-        if row["pipeline_name"] == "Name_dict":
-            assert row["parameters"]["Imputer"]["numeric_impute_strategy"] in {
-                "most_frequent",
-                "mean",
-            }
-            assert row["parameters"]["Imputer_1"]["numeric_impute_strategy"] in {
-                "median",
-                "mean",
-            }
-            assert row["parameters"]["Random Forest Classifier"]["n_estimators"] in {
-                50,
-                100,
-            }
-
-
 def test_automl_adds_pipeline_parameters_to_custom_pipeline_hyperparams(
     AutoMLTestEnv, X_y_binary
 ):
     X, y = X_y_binary
-
-    component_graph = {
-        "Imputer": ["Imputer", "X", "y"],
-        "Imputer_1": ["Imputer", "Imputer.x", "y"],
-        "One Hot Encoder": ["One Hot Encoder", "Imputer_1.x", "y"],
-        "Random Forest Classifier": [
-            "Random Forest Classifier",
-            "One Hot Encoder.x",
-            "y",
-        ],
+    pipeline_parameters = {"Imputer": {"numeric_impute_strategy": "most_frequent"}}
+    custom_hyperparameters = {
+        "One Hot Encoder": {"top_n": Categorical([12, 10])},
+        "Imputer": {
+            "numeric_impute_strategy": Categorical(["median", "most_frequent"])
+        },
     }
 
     automl = AutoMLSearch(
         X,
         y,
         problem_type="binary",
-        allowed_component_graphs={
-            "Pipeline One": component_graph,
-            "Pipeline Two": component_graph,
-            "Pipeline Three": component_graph,
-        },
         pipeline_parameters={"Imputer": {"numeric_impute_strategy": "most_frequent"}},
-        custom_hyperparameters={
-            "One Hot Encoder": {"top_n": Categorical([12, 10])},
-            "Imputer": {
-                "numeric_impute_strategy": Categorical(["median", "most_frequent"])
-            },
-        },
+        custom_hyperparameters=custom_hyperparameters,
         optimize_thresholds=False,
         max_batches=4,
     )
-    env = AutoMLTestEnv("binary")
-    with env.test_context(score_return_value={automl.objective.name: 1.767}):
-        automl.search()
-
-    for i, row in automl.full_rankings.iterrows():
-        if "Mode Baseline Binary" in row["pipeline_name"]:
-            continue
-        assert row["parameters"]["Imputer"]["numeric_impute_strategy"] in [
-            "most_frequent",
-            "median",
-        ]
-        assert 10 <= row["parameters"]["One Hot Encoder"]["top_n"] <= 12
+    assert automl._automl_algorithm._custom_hyperparameters == custom_hyperparameters
```

Comment:
Removed the `allowed_component_graph` parameter and checked through the algorithms `_pipeline_params` instead. Here is a place where I believe brunt of this test should be the algorithm tests responsibility (as we are delegating pipeline creation to the algorithms).